qoqo-iqm/src/backend.rs

```diff
@@ -0,0 +1,187 @@
+        self.clone()
+    }
+
+    /// Return the bincode representation of the Backend using the [bincode] crate.
+    ///
+    /// Returns:
+    ///     ByteArray: The serialized Backend (in [bincode] form).
+    ///
+    /// Raises:
+    ///     ValueError: Cannot serialize Backend to bytes.
+    pub fn to_bincode(&self) -> PyResult<Py<PyByteArray>> {
+        let serialized = serialize(&self.internal)
+            .map_err(|_| PyValueError::new_err("Cannot serialize Backend to bytes"))?;
+        let b: Py<PyByteArray> = Python::with_gil(|py| -> Py<PyByteArray> {
+            PyByteArray::new(py, &serialized[..]).into()
+        });
+        Ok(b)
+    }
+
+    /// Convert the bincode representation of the Backend to a Backend using the [bincode] crate.
+    ///
+    /// Args:
+    ///     input (ByteArray): The serialized Backend (in [bincode] form).
+    ///
+    /// Returns:
+    ///     Backend: The deserialized Backend.
+    ///
+    /// Raises:
+    ///     TypeError: Input cannot be converted to byte array.
+    ///     ValueError: Input cannot be deserialized to Backend.
+    #[staticmethod]
+    pub fn from_bincode(input: &PyAny) -> PyResult<BackendWrapper> {
+        let bytes = input
+            .extract::<Vec<u8>>()
+            .map_err(|_| PyTypeError::new_err("Input cannot be converted to byte array"))?;
+
+        Ok(BackendWrapper {
+            internal: deserialize(&bytes[..])
+                .map_err(|_| PyValueError::new_err("Input cannot be deserialized to Backend"))?,
+        })
+    }
+
+    /// Run a circuit with the IQM backend.
+    ///
+    /// A circuit is passed to the backend and executed.
+    /// During execution values are written to and read from classical registers
+    /// (List[bool], List[float], List[complex]).
+    /// To produce sufficient statistics for evaluating expectation values,
+    /// circuits have to be run multiple times.
+    /// The results of each repetition are concatenated in OutputRegisters
+    /// (List[List[bool]], List[List[float]], List[List[complex]]).  
+    ///
+    /// Args:
+    ///     circuit (Circuit): The circuit that is run on the backend.
+    ///
+    /// Returns:
+    ///     Tuple[Dict[str, List[List[bool]]], Dict[str, List[List[float]]]], Dict[str, List[List[complex]]]]: The output registers written by the evaluated circuits.
+    ///
+    /// Raises:
+    ///     TypeError: Circuit argument cannot be converted to qoqo Circuit
+    ///     RuntimeError: Running Circuit failed
+    pub fn run_circuit(&self, circuit: &PyAny) -> PyResult<Registers> {
+        let circuit = convert_into_circuit(circuit).map_err(|err| {
+            PyTypeError::new_err(format!(
+                "Circuit argument cannot be converted to qoqo Circuit {:?}",
+                err
+            ))
+        })?;
+        self.internal
+            .run_circuit(&circuit)
+            .map_err(|err| PyRuntimeError::new_err(format!("Running Circuit failed {:?}", err)))
+    }
+}
```

Comment:
We also need to add the run_measurement_registers and run_measurement functions in here. See for example qoqo_aqt.